src/Shared/ThrowHelpers/ArgumentNullThrowHelper.cs

```diff
@@ -26,8 +30,10 @@ public static void ThrowIfNull([NotNull] object? argument, [CallerArgumentExpres
 #endif
     }
 
-#if !NET7_0_OR_GREATER
+#if !NET7_0_OR_GREATER || NETSTANDARD || NETFRAMEWORK
```

Comment:
doesn't `#if !NET7_0_OR_GREATER` include NETSTANDARD  and NETFRAMEWORK? Assuming SDK style projects.

Reply:
From my understanding of [the docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#conditional-compilation) and the errors I was seeing in the build that resulted in this fix `NET7_0_OR_GREATER` is only set for projects that target a .NET Core version. They aren't set at all for netstandard or netfx projects.

Reply:
I don't understand. If `NET7_0_OR_GREATER` is **not** defined then you're in netstandard or netframework, which is what we want. And that's what `#if !NET7_0_OR_GREATER` does.